modules/cloud-init/nfs-cloud-init.tftpl

```diff
@@ -8,15 +8,31 @@ users:
       - ${ssh_public_key}
 
 runcmd:
+# install nfs server package
 - apt-get update
 - apt-get install -y nfs-kernel-server
+
+# Prepare partition on secondary disk
 - mkdir -p ${nfs_path}
-- mkfs.ext4 /dev/vda
-- mount -o rw /dev/vda ${nfs_path}
-- echo "/dev/vdb ${nfs_path} ext4  rw,relatime" | tee -a /etc/fstab
+- parted -s /dev/disk/by-id/virtio-${nfs_disk_id} mklabel gpt
+- parted -s /dev/disk/by-id/virtio-${nfs_disk_id} mkpart primary ext4 0% 100%
+- mkfs.ext4 /dev/disk/by-id/virtio-${nfs_disk_id}-part1
+- sync
+
+# Mount the partition
+- mount -o rw /dev/disk/by-id/virtio-${nfs_disk_id}-part1 ${nfs_path}
+- echo "/dev/disk/by-id/virtio-${nfs_disk_id}-part1 ${nfs_path} ext4 rw,relatime" >> /etc/fstab
```

Comment:
This better to be replaced with `disk_setup`, `fs_setup` and `mounts` clauses for better maintainability. See the cloud-init reference manual https://cloudinit.readthedocs.io/en/latest/reference/modules.html#disk-setup

Just use `device_aliases` to setup config alias for `/dev/disk/by-id/` and then use this alias to configure partitions and filesystems

Reply:
[update to disk_setup, fs_setup, and mounts in cloud-init]
(https://github.com/nebius/nebius-solution-library/pull/120/commits/39e7d3c6936f58fa45cd7ae9a6e91d01965ccc0b)